### PR TITLE
Add traefik.domain label #51

### PR DESCRIPTION
**Pull request by EmileVauge**
This PR adds a traefik.domain label for docker / marathon providers.

### CODE REVIEW

File: docker.go [hunk 1]
```diff
@@ -105,6 +71,47 @@ func (provider *DockerProvider) Provide(configurationChan chan<- configMessage)
 }
 
 func (provider *DockerProvider) loadDockerConfig(dockerClient *docker.Client) *Configuration {
+	var DockerFuncMap = template.FuncMap{
```
Comment (vdemeester) [inline docker.go]: Any reason for the move of this var ? :smile_cat:

Comment (EmileVauge) [inline docker.go]: Yep, because I needed to access to provider ;)

Comment (vdemeester) [inline docker.go]: oh right, didn't look too closely :sweat:

**Comment by vdemeester** [comment]
@EmileVauge needs a rebase :stuck_out_tongue_closed_eyes:

**Comment by InAnimaTe** [comment]
Looks good to me. Any reason you only did this for those two providers? I feel like this option should be something always available for defining on any container no matter what provider you use (not demanding or anything, just suggesting).

**Comment by EmileVauge** [comment]
Hi @InAnimaTe, other providers don't need it because you can directly put whatever domain you want in the Key Values store. You can refer to the documentation https://github.com/EmileVauge/traefik/blob/master/docs/index.md#-etcd-backend, there is examples of Traefik configuration using these providers :)

**Comment by InAnimaTe** [comment]
Ahh yes, you are correct. I should learn to read next time ;)

Nice work on this and I appreciate the quick turn-around. I plan on testing it as soon a docker image is available.

**Comment by EmileVauge** [comment]
Once [CircleCI](https://circleci.com/gh/EmileVauge/traefik/177) is green and @vdemeester gives his LGTM, you will get it :)

**Comment by vdemeester** [comment]
LGTM 🐶

**Pull request merged**